### PR TITLE
[System.Net] Replace use of obsolete `SupportsIPv6` property.

### PR DESCRIPTION
**Pull request by alexischr**
The obsolete property was always set to false. Fixes https://github.com/mono/mono/issues/6940

### CODE REVIEW

**Comment by akoeplinger** [comment]
The codepath quoted on the issue which always sets the obsolete property to false isn't used on Mono so I don't understand why this happens.

**Comment by alexischr** [comment]
@akoeplinger we define `FEATURE_PAL`: https://github.com/mono/mono/blob/master/mcs/class/System/Makefile#L62

**Comment by marek-safar** [comment]
I think the fix should be to move the code we have to location which is compiled 
https://github.com/mono/mono/blob/4e3389667f6e29480cf978eb524d0c4308fe6cb5/mcs/class/referencesource/System/net/System/Net/Sockets/Socket.cs#L6065

to

https://github.com/mono/mono/blob/4e3389667f6e29480cf978eb524d0c4308fe6cb5/mcs/class/referencesource/System/net/System/Net/Sockets/Socket.cs#L6129

**Comment by akoeplinger** [comment]
@alexischr you missed this little nugget at the top of the file 😄 :
https://github.com/mono/mono/blob/4e3389667f6e29480cf978eb524d0c4308fe6cb5/mcs/class/referencesource/System/net/System/Net/Sockets/Socket.cs#L7-L9

**Comment by alexischr** [comment]
@akoeplinger  Sorry, I saw that today 🤦‍♂️ 

That got me confused and had to verify that the change in `Dns.cs` fixes the behavior (it does, and the other change is not needed). I have to figure how `OSSupportsIPv6` can possibly be true when `SupportsIPv6` apparently wasn't.

**Comment by alexischr** [comment]
oh: https://github.com/mono/mono/blob/4e3389667f6e29480cf978eb524d0c4308fe6cb5/mcs/class/referencesource/System/net/System/Net/Sockets/Socket.cs#L6117

**Comment by akoeplinger** [comment]
Yeah but that should just return true if the config section isn't there: https://github.com/mono/mono/blob/4e3389667f6e29480cf978eb524d0c4308fe6cb5/mcs/class/System/ReferenceSources/SettingsSectionInternal.cs#L31-L43

... and looking at our machine.config file we're not setting it.

**Comment by alexischr** [comment]
@akoeplinger "system.net/settings" section is declared here: https://github.com/mono/mono/blob/29c2ba9387fe094c672adedd27c56e46bfa3669c/data/net_4_5/machine.config#L87

And the "ipv6/enabled" defaults to false: https://github.com/mono/mono/blob/29c2ba9387fe094c672adedd27c56e46bfa3669c/mcs/class/System/System.Net.Configuration/Ipv6Element.cs#L65

File: mcs/class/referencesource/System/net/System/Net/Internal.cs [hunk 1]
```diff
@@ -318,7 +318,7 @@ private static IPHostEntry GetLocalHost()
             //               old IPv4 gethostbyname(null). Instead we need
             //               to do a more complete lookup.
             //
-            if (Socket.SupportsIPv6)
+            if (Socket.OSSupportsIPv6)
```
Comment (akoeplinger) [inline mcs/class/referencesource/System/net/System/Net/Internal.cs]: please revert this change, we're not using this code.

File: mcs/class/System/System.Net/Dns.cs [hunk 2]
```diff
@@ -320,11 +320,9 @@ private static IPHostEntry hostent_to_IPHostEntry(string originalHostName, strin
 				try {
 					IPAddress newAddress = IPAddress.Parse(h_addrlist[i]);
 
-#pragma warning disable 618
-					if( (Socket.SupportsIPv6 && newAddress.AddressFamily == AddressFamily.InterNetworkV6) ||
+					if( (Socket.OSSupportsIPv6 && newAddress.AddressFamily == AddressFamily.InterNetworkV6) ||
 					    (Socket.SupportsIPv4 && newAddress.AddressFamily == AddressFamily.InterNetwork) )
```
Comment (akoeplinger) [inline mcs/class/System/System.Net/Dns.cs]: please use `Socket.OSSupportsIPv4` here so we're consistent.

**Comment by marek-safar** [comment]
@monojenkins backport 2017-12

**Comment by marek-safar** [comment]
@monojenkins backport 2018-02

**Comment by marek-safar** [comment]
@alexischr please backport manually
